cardano-cli/test/cardano-cli-test/Test/Cli/Parser.hs

```diff
@@ -0,0 +1,105 @@
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications #-}
+
+module Test.Cli.Parser
+  ( hprop_integral_reader
+  , hprop_integral_pair_reader_positive
+  , hprop_integral_pair_reader_negative
+  )
+where
+
+import           Cardano.CLI.EraBased.Options.Common (integralParsecParser,
+                   pairIntegralParsecParser)
+
+import           Data.Bits (Bits)
+import           Data.Data (Proxy (..), Typeable)
+import           Data.Either (isLeft, isRight)
+import           Data.Word (Word16)
+import qualified Text.Parsec as Parsec
+
+import           Hedgehog (Gen, Property, assert, property)
+import           Hedgehog.Extras (propertyOnce)
+import qualified Hedgehog.Gen as Gen
+import           Hedgehog.Internal.Property (forAll)
+import qualified Hedgehog.Range as Range
+
+-- | Execute me with:
+-- @cabal test cardano-cli-test --test-options '-p "/integral reader/"'@
+hprop_integral_reader :: Property
+hprop_integral_reader = propertyOnce $ do
+  assert $ isRight $ parse @Word "0"
```

Comment:
You should use https://hackage.haskell.org/package/hedgehog-1.5/docs/Hedgehog-Internal-Gen.html#v:word instead.

Reply:
:+1: Done with:

```haskell
  w <- forAll $ Gen.word $ Gen.linear minBound maxBound
  assert $ isRight $ parse @Word (show w)
```